cli/azd/pkg/project/service_target_aks.go

```diff
@@ -0,0 +1,444 @@
+}
+
+// Finds a deployment using the specified deploymentNameFilter string
+// Waits until the deployment rollout is complete nad all replicas are accessible
+func (t *aksTarget) waitForDeployment(
+	ctx context.Context,
+	namespace string,
+	deploymentNameFilter string,
+) (*kubectl.Deployment, error) {
+	return kubectl.WaitForResource(
+		ctx, t.kubectl, namespace, kubectl.ResourceTypeDeployment,
+		func(deployment *kubectl.Deployment) bool {
+			return strings.Contains(deployment.Metadata.Name, deploymentNameFilter)
+		},
+		func(deployment *kubectl.Deployment) bool {
+			return deployment.Status.AvailableReplicas == deployment.Spec.Replicas
+		},
+	)
+}
+
+// Finds an ingress using the specified ingressNameFilter string
+// Waits until the ingress LoadBalancer has assigned a valid IP address
+func (t *aksTarget) waitForIngress(
+	ctx context.Context,
+	namespace string,
+	ingressNameFilter string,
+) (*kubectl.Ingress, error) {
+	return kubectl.WaitForResource(
+		ctx, t.kubectl, namespace, kubectl.ResourceTypeIngress,
+		func(ingress *kubectl.Ingress) bool {
+			return strings.Contains(ingress.Metadata.Name, ingressNameFilter)
+		},
+		func(ingress *kubectl.Ingress) bool {
+			var ipAddress string
+			for _, config := range ingress.Status.LoadBalancer.Ingress {
+				if config.Ip != "" {
+					ipAddress = config.Ip
+					break
+				}
+			}
+
+			return ipAddress != ""
+		},
+	)
+}
+
+// Finds a service using the specified serviceNameFilter string
+// Waits until the service is available
+func (t *aksTarget) waitForService(
+	ctx context.Context,
+	namespace string,
+	serviceNameFilter string,
+) (*kubectl.Service, error) {
+	return kubectl.WaitForResource(
+		ctx, t.kubectl, namespace, kubectl.ResourceTypeService,
+		func(service *kubectl.Service) bool {
+			return strings.Contains(service.Metadata.Name, serviceNameFilter)
+		},
+		func(service *kubectl.Service) bool {
+			// If the service is not a load balancer it should be immediately available
+			if service.Spec.Type != kubectl.ServiceTypeLoadBalancer {
+				return true
+			}
+
+			// Load balancer can take some time to be provision by AKS
+			var ipAddress string
+			for _, config := range service.Status.LoadBalancer.Ingress {
+				if config.Ip != "" {
+					ipAddress = config.Ip
+					break
+				}
+			}
+
+			return ipAddress != ""
+		},
+	)
+}
+
+// Retrieve any service endpoints for the specified namespace and serviceNameFilter
+// Supports service types for LoadBalancer and ClusterIP
+func (t *aksTarget) getServiceEndpoints(ctx context.Context, namespace string, serviceNameFilter string) ([]string, error) {
+	service, err := t.waitForService(ctx, namespace, serviceNameFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	var endpoints []string
+	if service.Spec.Type == kubectl.ServiceTypeLoadBalancer {
+		for _, resource := range service.Status.LoadBalancer.Ingress {
+			endpoints = append(endpoints, fmt.Sprintf("http://%s (Service, Type: LoadBalancer)", resource.Ip))
```

Comment:
I think this might break `vscode` relying on `endpoints` being an actual URI. We might need to introduce structured endpoints to support this.

Reply:
Do we need to show load balancer's ip?  What can I do with this as a customer?

Reply:
@vhvb1989 The Load balance IP would typically be the public IP address assigned to the load balancer and this would be the URL that you would use to access your application.  Unlike many other Azure services it does not include a built in DNS endpoint without a custom configuration + TLS.

For this simple sample you would access the IP address over HTTP in your browser.

Reply:
somehow I thought that there would be one url-endpoint for each service/container, and when we try to access any of those, the request would go first to the `load balancer` and then to the actual service.
But I just remembered now that you explained that there is a manifest where we need to create the mapping from the main endpoint/app to the actual container/service.  Now I understand that `main endpoint` is the load balancer!  Thank you!.